scripts/check-flatc.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+#   Copyright The containerd Authors.
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#       http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+set -eux -o pipefail
+
+CUR_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+SOCI_SNAPSHOTTER_PROJECT_ROOT="${CUR_DIR}/.."
+FBS_FILE_PATH=${SOCI_SNAPSHOTTER_PROJECT_ROOT}/ztoc/fbs/ztoc.fbs
+
+# check if flatbuffers needs to be generated again
+TMPDIR=$(mktemp -d)
+flatc -o ${TMPDIR} -g ${FBS_FILE_PATH}
+diff -qr ${TMPDIR}/ztoc ${SOCI_SNAPSHOTTER_PROJECT_ROOT}/ztoc/fbs/ztoc || (printf "\n\nThe Ztoc schema seems to be modified. Please run 'make flatc' to re-generate Go files\n\n"; exit 1)
+rm -rf ${TMPDIR}
```

Comment:
This temp dir doesn't get deleted if the check fails. Looks like that was a problem when it was in the makefile too, but we should probably fix it.

Reply:
Add the `rm` call before `exit 1` so it's deleted and script exits with 1.

Reply:
This was intentional. The idea (from wmesard@) was that users might be interested in what changed if the check fails.

Reply:
The script will print out the diff if there is any. In most case I think the user will need to run `make flatc` (which will also shows the diff in git). I think it should be sufficient, but I don't have a strong preference since it's just a tmp folder :)